File: nomenclator/src/main/java/flowmanager/nomenclator/model/Team.java
package flowmanager.nomenclator.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(exclude = {
        "organization",
        "manager",
        "projects",
        "members"
})
@Builder
@Entity
@Table(name = "team")
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "organization_id")
    private Organization organization;

    @ManyToOne
    @JoinColumn(name = "manager_id")
    private User manager;

    @Builder.Default
    @ManyToMany
    @JoinTable(
            name = "team_project",
            joinColumns = @JoinColumn(name = "team_id"),
            inverseJoinColumns = @JoinColumn(name = "project_id")
    )
    private List<Project> projects = new ArrayList<>();

    @Builder.Default
    @ManyToMany
    @JoinTable(
            name = "team_user",
            joinColumns = @JoinColumn(name = "team_id"),
            inverseJoinColumns = @JoinColumn(name = "user_id")
    )
    private List<User> members = new ArrayList<>();
}

File: frontend/app/types/team.ts
import type { UserSummaryDto } from "./user"
import type { OrganizationSummaryDto } from "./organization"
import type { ProjectSummaryDto } from "./project"

export interface TeamCreateDto {
  name: string
  description: string
  organizationId: number
  membersIds?: number[]
}

export interface TeamUpdateDto {
  name?: string
  description?: string
  managerId?: number
  membersIds?: number[]
}

export interface TeamSummaryOrganizationDto {
  id: number
  name: string
  manager: UserSummaryDto
}

export interface TeamSummaryUserDto {
  id: number
  name: string
  organization: OrganizationSummaryDto
  createdAt: string
}

export interface TeamSummaryDto {
  id: number
  name: string
  description: string
  organization: OrganizationSummaryDto
  manager: UserSummaryDto
}

export interface TeamResponseDto {
  id: number
  name: string
  description: string
  createdAt: string
  organization: OrganizationSummaryDto
  manager: UserSummaryDto
  projects: ProjectSummaryDto[]
  members: UserSummaryDto[]
}